benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/RedisProtocolBenchmark.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.AspNetCore.SignalR.Redis.Internal;
+
+namespace Microsoft.AspNetCore.SignalR.Microbenchmarks
+{
+    public class RedisProtocolBenchmark
+    {
+        private RedisProtocol _protocol;
+        private RedisGroupCommand _groupCommand;
+        private object[] _args;
+        private string _methodName;
+        private IReadOnlyList<string> _excludedIdsSmall;
+        private IReadOnlyList<string> _excludedIdsLarge;
+        private byte[] _writtenAck;
+        private byte[] _writtenGroupCommand;
+        private byte[] _writtenInvocationNoExclusions;
+        private byte[] _writtenInvocationSmallExclusions;
+        private byte[] _writtenInvocationLargeExclusions;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            _protocol = new RedisProtocol(new [] {
+                new DummyProtocol("protocol1"),
+                new DummyProtocol("protocol2")
+            });
+
+            _groupCommand = new RedisGroupCommand(id: 42, serverName: "Server", GroupAction.Add, groupName: "group", connectionId: "connection");
+
+            // Because of the DummyProtocol, the args don't really matter
+            _args = Array.Empty<object>();
+            _methodName = "Method";
+
+            _excludedIdsSmall = GenerateIds(2);
+            _excludedIdsLarge = GenerateIds(20);
+
+            _writtenAck = _protocol.WriteAck(42);
+            _writtenGroupCommand = _protocol.WriteGroupCommand(_groupCommand);
+            _writtenInvocationNoExclusions = _protocol.WriteInvocation(_methodName, _args, null);
+            _writtenInvocationSmallExclusions = _protocol.WriteInvocation(_methodName, _args, _excludedIdsSmall);
+            _writtenInvocationLargeExclusions = _protocol.WriteInvocation(_methodName, _args, _excludedIdsLarge);
```

Comment:
Couldn't use Params because these variations only apply to one benchmark. I thought this was cleaner than splitting into yet another class.